### PR TITLE
[pdata] Rename enum zero value constants to have `Empty` suffix

### PR DESCRIPTION
**Pull request by dmitryax**
For consistency with `pcommon.ValueType`

Resolves https://github.com/open-telemetry/opentelemetry-collector/issues/6256

### CODE REVIEW

**Review by bogdandrutu — COMMENTED** [review]
I don't think we should do this to be honest. In case of "Value", the `Empty` has a real value (is correct), in case for `None` that is not really valid.

**Comment by dmitryax** [comment]
Now I see that I misread you comment in https://github.com/open-telemetry/opentelemetry-collector/issues/6256. Do you think we should keep it as `None` or `Unspecified`?

**Comment by bogdandrutu** [comment]
@dmitryax let's think a bit more about this. "empty" may not be bad option, since it expose that none of the values within the `oneof` group is present, aka the `oneof` is empty. 
* For `MetricType`, empty sounds good, though the metric may have a name (is not completely empty).
*  `NumberDataPointValueType` , empty sounds maybe that the value is `empty`?

I can be convinced to do this change, does not necessary seem the worst, since from protocol perspective we have the capability to really on the presence of the oneof (any member) or not.

**Comment by dmitryax** [comment]
> "empty" may not be bad option, since it expose that none of the values within the oneof group is present, aka the oneof is empty.

That was my original idea.

> For MetricType, empty sounds good, though the metric may have a name (is not completely empty).

I believe metric with a name still can be "empty". It sounds more natural for me than `None` metric type. Maybe `Unspecified` is a good choice as well.

> NumberDataPointValueType , empty sounds maybe that the value is empty?

It still mostly applied to Value _Type_ rather value itself, but I believe it's ok to interpret it like that.

Both `Empty` and `Unspecified` sound like a good options for me, not sure about `None` tho. Maybe `Empty` is better for `MetricType` and `Unspecified` for `NumberDataPointValueType` and `ExemplarValueType`.

File: pdata/pmetric/metrics.go [hunk 1]
```diff
@@ -102,19 +102,22 @@ func (ms Metrics) DataPointCount() (dataPointCount int) {
 type MetricType int32
 
 const (
-	MetricTypeNone MetricType = iota
+	MetricTypeEmpty MetricType = iota
```
Comment (bogdandrutu) [inline pdata/pmetric/metrics.go:105]: Add a comment to the value that it means none of the "possible" values is set? Or something in that lines.

**Review by bogdandrutu — APPROVED** [review]

File: pdata/pmetric/metrics.go (hunk 1 above)
Comment (dmitryax) [reply pdata/pmetric/metrics.go:106]: Added a comment to these

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**